sklearn/lda.py

```diff
@@ -231,11 +231,19 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
                  n_components=None, store_covariance=False, tol=1e-4):
         self.solver = solver
         self.shrinkage = shrinkage
-        self.priors = priors
         self.n_components = n_components
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+        self.priors = np.asarray(priors) if priors is not None else None
+
+        if self.priors is not None:
+            if (self.priors < 0).any():
+                raise ValueError('priors must be non-negative')
+            if self.priors.sum() != 1:
+                print('warning: the priors do not sum to 1. Renormalizing')
+                self.priors = self.priors / self.priors.sum()
```

Comment:
No conversion should ever happen in init. All should be done in fit, and self.priors should not be changed. Please see code guidelines.